### ISSUE TITLE
[java] EqualsAvoidNull

### ISSUE THREAD

**Issue by Pankraz76** [pmd/pmd#5896]
**EqualsAvoidNull:**

- https://checkstyle.sourceforge.io/checks/coding/equalsavoidnull.html
- https://github.com/junit-team/junit-framework/pull/4748
- https://github.com/junit-team/junit-framework/commit/aca30c5b64805a424b6639ec9ceecc23a5cd5c10

As discovered in: 
- https://github.com/Pankraz76/pmd/pull/1

these rule might be absent due to the dogfood approach not covering these changes.

**Proposed Category:** One of [Best Practices | Code Style | Design | Documentation | Error Prone | Multithreading | Performance | Security]

**Description:**

**Code Sample:** This should include code, that should be flagged by the rule. If possible, the "correct" code
according to this new rule should also be demonstrated.

```java
public class Example1 {
  public void foo() {
    String nullString = null;

    // violation below 'String literal expressions should be on the left side'
    nullString.equals("My_Sweet_String");
    "My_Sweet_String".equals(nullString);
    // violation below 'String literal expressions should be on the left side'
    nullString.equalsIgnoreCase("My_Sweet_String");
    "My_Sweet_String".equalsIgnoreCase(nullString);
  }
}
```

**Possible Properties:**

*   Should this rule be customizable via properties?

**Comment by oowekyala** [comment]
This is the rule [LiteralsFirstInComparisons](https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#literalsfirstincomparisons). What do you mean "due to the dogfood approach not covering these changes"?

**Comment by Pankraz76** [comment]
yes of course i mean LiteralsFirstInComparisons im sorry. Assuming then there is a blindspot as there are changes beging made by the rewrite variant of it.

- https://github.com/Pankraz76/pmd/pull/1

**Comment by oowekyala** [comment]
There is nothing wrong with the rule. These changes are in test sources, which are not checked with that rule. Same thing as #5738.
